src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -1,690 +1,687 @@
       }
+    }
+  },
 
-      if (this.selectedHintIndex == nextActiveHintIndex) return;
+  getListNode(index) {
+    return this.listContainer.childNodes[index];
+  },
 
-      let node = this.listContainer.childNodes[this.selectedHintIndex];
+  changeActive(nextActiveHintIndex, avoidWrap) {
+    let nextIndex = nextActiveHintIndex;
 
-      if (node) {
-        node.className = node.className.replace(' ' + ACTIVE_HINT_ELEMENT_CLASS, '');
-      }
+    if (nextIndex >= this.itemsOptions.length) {
+      nextIndex = avoidWrap ? this.itemsOptions.length - 1 : 0;
+    } else if (nextIndex < 0) {
+      nextIndex = avoidWrap ? 0 : this.itemsOptions.length - 1;
+    }
 
-      this.selectedHintIndex = nextActiveHintIndex;
-      node = this.listContainer.childNodes[this.selectedHintIndex];
-      node.className += ' ' + ACTIVE_HINT_ELEMENT_CLASS;
+    if (this.selectedHintIndex === nextIndex) return;
 
-      if (node.offsetTop < this.listContainer.scrollTop) {
-        this.listContainer.scrollTop = node.offsetTop - 3;
-      } else if (node.offsetTop + node.offsetHeight > this.listContainer.scrollTop + this.listContainer.clientHeight) {
-        this.listContainer.scrollTop = node.offsetTop + node.offsetHeight - this.listContainer.clientHeight + 3;
-      }
-    },
-  };
+    let node = this.listContainer.childNodes[this.selectedHintIndex];
 
-  function createHeader(text, className = '') {
-    const header = createInitDiv(className);
+    if (node) {
+      node.className = node.className.replace(` ${ACTIVE_HINT_ELEMENT_CLASS}`, '');
+    }
 
-    header.innerHTML = text;
+    this.selectedHintIndex = nextIndex;
+    node = this.listContainer.childNodes[this.selectedHintIndex];
+    node.className += ` ${ACTIVE_HINT_ELEMENT_CLASS}`;
 
-    return header;
-  }
+    if (node.offsetTop < this.listContainer.scrollTop) {
+      this.listContainer.scrollTop = node.offsetTop - 3;
+    } else if (node.offsetTop + node.offsetHeight > this.listContainer.scrollTop + this.listContainer.clientHeight) {
+      this.listContainer.scrollTop = node.offsetTop + node.offsetHeight - this.listContainer.clientHeight + 3;
+    }
+  },
+};
 
-  function createInitDiv(className = '') {
-    const divElement = document.createElement('div');
+function applicableHelpers(cm, helpers) {
+  if (!cm.somethingSelected()) return helpers;
 
-    divElement.className = className;
+  return helpers.reduce((memo, helper) => helper.supportsSelection ? memo.concat(helper) : memo, []);
+}
 
-    return divElement;
-  }
+function fetchHints(hint, cm, options, callback) {
+  if (hint.async) {
+    hint(cm, callback, options);
+  } else {
+    const result = hint(cm, options);
 
-  function applicableHelpers(cm, helpers) {
-    if (!cm.somethingSelected()) return helpers;
-    const result = [];
-    for (let i = 0; i < helpers.length; i++) if (helpers[i].supportsSelection) result.push(helpers[i]);
-    return result;
+    if (result && result.then) result.then(callback);
```

Comment:
Please format it

```
result && result.then ? result.then(callback) : callback(result);
```

Reply:
This won't pass `eslint` rule which requires assigning expression. So I will leave as is.